src/eterna/folding/EternafoldThreshknot.ts

```diff
@@ -0,0 +1,172 @@
+                    const base2 = bpp[index + 1];
+                    const prob = bpp[index + 2];
+                    if (prob === Pmax[base1 - 1] && prob === Pmax[base2 - 1] && prob > _theta) {
+                        current_bpList.push([base1, base2]);
+                    }
+                }
+                // Update iteration flags
+                newBP = current_bpList.length;
+                iteration += 1;
+                if (newBP !== 0 && iteration > maxIterations) {
+                    log.debug('Reached max iteration, stopping before converged.');
+                } else {
+                // Add selected base pairs to output list
+                    bpList.push(...current_bpList);
+                }
+            }
+
+            // Ensure that the bpList is sorted, then check for duplicated nucleotides
+            bpList.sort((bpA, bpB) => bpA[0] - bpB[0]);
+            const nts = bpList.flat();
+            if (nts.length > Array.from(new Set(nts)).length) {
+                log.warn('Some nucletotides found in more than 1 base pair');
+                bpList.forEach((bpA, i) => {
+                    const bpB = bpList[i + 1] || [];
+                    if (bpA[0] === bpB[0] && bpA[1] === bpB[1]) {
+                        log.warn(`Removing duplicated base pair: ${bpA}`);
+                        bpList.splice(i, 1);
+                    } else if (bpB.includes(bpA[0])) {
+                        log.warn(`bpA: ${bpA}, bpB: ${bpB}`);
+                    } else if (bpB.includes(bpA[1])) {
+                        log.warn(`bpA: ${bpA}, bpB: ${bpB}`);
+                    }
+                });
+            }
+
+            // Convert from array of pair tuples to partner pair list
+            const partnerPairList = new Array(seqStr.length).fill(-1);
+            bpList.forEach((pair) => {
+                // Indices are 1-indexed here; Dot Plot returns bp 1-indexed
+                partnerPairList[pair[0] - 1] = pair[1] - 1;
+                partnerPairList[pair[1] - 1] = pair[0] - 1;
+            });
+
+            // Convert to SecStruct and remove single pair helices
+            const structure = new SecStruct(partnerPairList);
+            const helices = structure.stems();
+            const pruned_helices = helices.filter((stem) => stem.length >= minLenHelix).flat();
+
+            // Convert from array of pair tuples to partner pair list
+            const pruned_pair_list = new Array(structure.length).fill(-1);
+            pruned_helices.forEach((pair) => {
+                // Indices are 0-indexed here
+                pruned_pair_list[pair[0]] = pair[1];
+                pruned_pair_list[pair[1]] = pair[0];
+            });
+
+            return new SecStruct(pruned_pair_list);
+        } catch (e) {
+            log.error('FullFoldTemperature error', e);
+            return new SecStruct();
+        } finally {
+            if (result != null) {
+                // result.delete();
```

Comment:
Why is the result not deleted? This will probably cause a memory leak, since it's a pointer to WASM memory that isn't GCed

Reply:
TS throws an error: `Property 'delete' does not exist on type 'never'`. But I can look into why. I didn't fully understand how the engines were interacting with the WASM module, but that makes sense.